server/event_handler.py

```diff
@@ -0,0 +1,137 @@
+from typing import cast
+
+from server.client import Client
+from server.codes import StatusCode
+from server.connection_manager import ConnectionManager
+from server.events import (
+    ConnectData,
+    ErrorData,
+    EventRequest,
+    EventResponse,
+    EventType,
+    MoveData,
+    ReplaceData,
+    SyncData,
+)
+from server.room import Room
+
+
+class EventHandler:
+    """An request event handler."""
+
+    def __init__(self, client: Client, manager: ConnectionManager):
+        """Initializes the event handler for each client.
+
+        Args:
+            client: The client sending the requests.
+            connection: The connection to the room.
+        """
+        self.client = client
+        self.manager = manager
+
+        self.room: Room
+
+    async def __call__(self, request: EventRequest, room_code: str) -> None:
+        """Handle a request received.
+
+        Args:
+            request: The data received from the client.
+            room_code: The room to which the data will be sent.
+
+        Raises:
+            WebSocketDisconnect: If the event type is a disconnect.
+            NotImplementedError: In any other case.
+        """
+        event_data = request.data
+
+        match request.type:
+            case EventType.CONNECT:
+                connect_data = cast(ConnectData, event_data)
+                self.client.username = connect_data.username
+
+                match connect_data.connection_type:
+                    case "create":
+                        if connect_data.difficulty is None:
+                            response = (
+                                EventResponse(
+                                    type=EventType.ERROR,
+                                    data=ErrorData(message="Data not found."),
+                                    status_code=StatusCode.DATA_NOT_FOUND,
+                                ),
+                            )
+                            await self.client.send(response)
+                            return
+
+                        self.manager.create_room(self.client, connect_data.room_code, connect_data.difficulty)
```

Comment:
A Sync EventResponse needs to be sent to the client joining so the frontend knows that they received the request.